Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py

```diff
@@ -19,16 +21,27 @@
 class WebServerBenchmarkRunner(BenchmarkRunner):
     name = 'webserver'
 
-    def __init__(self, plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit=True, show_iteration_values=False, device_id=None, diagnose_dir=None):
+    def __init__(self, plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit=True, show_iteration_values=False, device_id=None, diagnose_dir=None, profile_output_directory=None):
         self._http_server_driver = HTTPServerDriverFactory.create(platform)
         self._http_server_driver.set_device_id(device_id)
-        super(WebServerBenchmarkRunner, self).__init__(plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit, show_iteration_values, device_id, diagnose_dir)
+        self._profile_output_directory = profile_output_directory
+        super(WebServerBenchmarkRunner, self).__init__(plan_file, local_copy, count_override, build_dir, output_file, platform, browser, browser_path, scale_unit, show_iteration_values, device_id, diagnose_dir, profile_output_directory)
         if self._diagnose_dir:
             self._http_server_driver.set_http_log(os.path.join(self._diagnose_dir, 'run-benchmark-http.log'))
 
     def _get_result(self, test_url):
         result = self._browser_driver.add_additional_results(test_url, self._http_server_driver.fetch_result())
         assert(not self._http_server_driver.get_return_code())
+
+        if self._profile_output_directory:
+            _log.info("Getting benchmark profile results for {} and copying them to {}.".format(test_url, self._profile_output_directory))
+            copy_output = subprocess.Popen(r"""log stream --style json --color none | perl -mFile::Basename -mFile::Copy -nle 'if (m/<WEBKIT_LLVM_PROFILE>.*<BEGIN>(.*)<END>/) { (my $l = $1) =~ s/\\\//\//g; my $b = File::Basename::basename($l); my $d = """ + "\"" + self._profile_output_directory + """/$b"; print "Moving $l to $d"; File::Copy::move($l, $d); }'""", shell=True, bufsize=0)
+            time.sleep(1)
+            os.system("notifyutil -p com.apple.WebKit.profiledata")
+            time.sleep(7)
+            copy_output.kill()
+            time.sleep(1)
+            _log.debug("Hopefully the benchmark profile has finished writing to disk, moving on.")
```

Comment:
Seems more useful to log what the paths were and their size on disk, than this.

Reply:
Oh I see, the copying is done by that perl script. We could at least log the size of the resulting path we copy to

Reply:
The size will always be the same, even if it is only filled with zeros, so I'm not sure what this gets us. We do log the file name already.